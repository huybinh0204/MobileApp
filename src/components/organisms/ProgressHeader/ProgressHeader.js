import { useNavigation } from '@react-navigation/native';
import PropTypes from 'prop-types';
import React from 'react';
import { IconSvg, ProgressBar } from '_components/atoms';
import { ICONS } from '_constants';
import { normalize } from '_utilities/screen';
import { ActionButton, HeaderContainer, ProgressBarContainer } from './ProgressHeader.styles';

const ProgressHeader = ({
  step,
  numberOfSteps,
  onBackPress,
  onRightItemPress,
  rightItem,
  rightItemTestID,
  showBackButton,
}) => {
  const { goBack } = useNavigation();

  return (
    <HeaderContainer>
      {showBackButton && (
        <ActionButton onPress={onBackPress ?? goBack} side="left" testID="BackButton">
          <IconSvg icon={ICONS.backIcon} width={normalize(25)} height={normalize(25)} />
        </ActionButton>
      )}
      <ProgressBarContainer>
        <ProgressBar step={step} numberOfSteps={numberOfSteps} />
      </ProgressBarContainer>
      {rightItem && (
        <ActionButton onPress={onRightItemPress} side="right" testID={rightItemTestID}>
          <IconSvg icon={rightItem} width={normalize(25)} height={normalize(25)} />
        </ActionButton>
      )}
    </HeaderContainer>
  );
};

ProgressHeader.defaultProps = {
  numberOfSteps: 10,
  onBackPress: null,
  onRightItemPress: null,
  rightItem: null,
  rightItemTestID: null,
  showBackButton: true,
};

ProgressHeader.propTypes = {
  step: PropTypes.number.isRequired,
  numberOfSteps: PropTypes.number,
  onBackPress: PropTypes.func,
  onRightItemPress: PropTypes.func,
  rightItem: PropTypes.string,
  rightItemTestID: PropTypes.string,
  showBackButton: PropTypes.bool,
};

export default ProgressHeader;
