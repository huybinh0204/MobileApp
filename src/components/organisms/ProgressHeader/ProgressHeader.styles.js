import styled from '@emotion/native';
import { normalize } from '_utilities/screen';

export const HeaderContainer = styled.View({
  alignItems: 'center',
  justifyContent: 'center',
  height: normalize(60),
});

export const ProgressBarContainer = styled.View({
  width: '50%',
});

export const ActionButton = styled.TouchableOpacity(({ theme, side }) => ({
  position: 'absolute',
  [side]: theme.spacing.general,
  zIndex: 10,
}));
