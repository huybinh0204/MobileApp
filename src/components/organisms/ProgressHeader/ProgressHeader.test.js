import { render } from '@testing-library/react-native';
import React from 'react';
import Theme from '_utilities/Theme';
import ProgressHeader from './ProgressHeader';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({ goBack: jest.fn() }),
}));

describe('ProgressHeader', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <Theme>
        <ProgressHeader step={5} numberOfSteps={10} />
      </Theme>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
