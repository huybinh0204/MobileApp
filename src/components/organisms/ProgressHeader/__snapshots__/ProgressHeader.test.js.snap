// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressHeader should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 108.5,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 36,
        "opacity": 1,
        "position": "absolute",
        "zIndex": 10,
      }
    }
    testID="BackButton"
  >
    <SvgMock
      height={45.5}
      width={45.5}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "width": "50%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#F2F3F4",
            "borderColor": "#000000",
            "borderRadius": 10,
            "height": 9,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#3D3397",
            "borderRadius": 10,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "0%",
          }
        }
      />
    </View>
  </View>
</View>
`;
